### ISSUE TITLE
Hoping to adapt a few PRs addressing some bugs

### ISSUE THREAD

**Issue by madejejej** [aerospike/aerospike-client-ruby#119]
In 2021 at [Castle](https://github.com/castle) we worked with @ioquatix because we found a few bugs and stability issues with the Ruby client. Unfortunately at the time we did not submit most of our changes to upstream. Would you be willing to take a look at these PRs and adapt them to the current codebase in some form or another?

- https://github.com/castle/aerospike-client-ruby/pull/8
- https://github.com/castle/aerospike-client-ruby/pull/7
- https://github.com/castle/aerospike-client-ruby/pull/6
- https://github.com/castle/aerospike-client-ruby/pull/5
- https://github.com/castle/aerospike-client-ruby/pull/4

**Comment by khaf** [comment]
Of course. We'll take a look and integrate the reasonable changes.

**Comment by khaf** [comment]
@madejejej Are you still using the Ruby client with Aerospike server? May I ask what is your use case?

**Comment by madejejej** [comment]
Yeah, we're still using the Ruby client. We're in the fraud prevention industry and we're computing key-value aggregates in Aerospike
